addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2270,11 +2284,13 @@ var SnippetsMenu = Widget.extend({
      *        true: always allows,
      *        false: always forbid,
      *        string: specific type of forbidden sanitization
+     * @param {Boolean} [toInsertInline]
+     *        elements which are inline as the "s_badge" snippet for example
      * @param {Object} [selectorGrids = []]
      *        elements which are in grid mode and for which a grid dropzone
      *        needs to be inserted
      */
-    _activateInsertionZones($selectorSiblings, $selectorChildren, canBeSanitizedUnless, selectorGrids = []) {
+    _activateInsertionZones($selectorSiblings, $selectorChildren, canBeSanitizedUnless, toInsertInline, selectorGrids = []) {
```

Comment:
I think this should be refactored: instead of "toInsertInline" receive the element to be inserted and do the check on it. Instead of duplicating the logic outside of this function. Leaving like that for now but would be nice if someone can do it later (after ARD's work's merge).